staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go

```diff
@@ -197,3 +199,33 @@ func (a *Webhook) Dispatch(attr admission.Attributes) error {
 	}
 	return a.dispatcher.Dispatch(ctx, &versionedAttr, relevantHooks)
 }
+
+// GetConvertor returns the versioned convertor.
+func (a *Webhook) GetConvertor() *versioned.Convertor {
```

Comment:
these getter and setting seem to be strange. Are they to inject mock objects? Maybe we need a way to inject them on construction. Having these getter and setters is not very Go idiomatic IMO.

Reply:
Yes, most are for mock objects, except this Convertor. And we can not inject them on construction, because this generic webhook is not called directly by test case, but by mutating and validating plugins. Or we can expose these fields。